pkg/agent/agent.go

```diff
@@ -191,6 +202,8 @@ func (i *Initializer) initInterfaceStore() error {
 
 func (i *Initializer) Initialize() error {
 	klog.Info("Setting up node network")
+	config.TrafficEncapMode = i.trafficEncapMode
```

Comment:
Is there a way to inject the mode to structs which need it instead of using a global variable which could break the e encapsulation?

Reply:
It was in NodeConfig, JianJun suggested to make it global because it is technically not a node configuration. I am fine either way.

Reply:
@suwang48404 : sorry it might be my fault not to explain my thoughts clearly. I do not mean to save the parameter to a global variable (we definitely should not do that for OO design), but meant either make it a separate field of AgentInitializer (and pass it to other interfaces/funcs as a separate parameter too) or define a new struct for all global network parameters (encapMode, ipsecPSK, serviceCIDR, tunnelType, etc.) and make it a field of AgentInitializer (and pass to other interfaces/funcs).

Reply:
MHO, passing same parameter to variable client/module is not good approach. If we have many parameters from agent-config.conf, it means we must pass each of them to a bunch of clients/modules, pretty messy.

as they are immutable, it'd be better to define them once, and be accessed freely by module/clients

my 2 cents, Su

Reply:
That is good point and I agree. Let me open an issue and hope to address this holistically.
I hope that OK as the problem we are address preexists to this PR.

Reply:
@suwang48404 : I am glad to refactor it more later. However, for this PR, I would suggest at least to pass this separate parameter like other parameters, instead of using a global variable. The whole Antrea code has no global variable.

Reply:
Ok, passed encapMode to individual consumers.